docs/source/x_ext.rst

```diff
@@ -648,15 +646,34 @@ Commit interface
+  |                    |                        | Shall an offloaded instruction be committed or killed?                                                                       |
+  |                    |                        | If ``commit_valid`` is 1 and ``commit_kill`` is 0,  then the core guarantees that the offloaded instruction (``id``) and any |
+  |                    |                        | older (i.e. preceding) instructions are no longer speculative, will not get killed (e.g. due to misspeculation or an         |
+  |                    |                        | exception in a preceding instruction), and are allowed to be committed.                                                      |
+  |                    |                        | If ``commit_valid`` is 1 and ``commit_kill`` is 1, then the offloaded instruction (``id``) and any newer (i.e. succeeding)   |
+  |                    |                        | instructions shall be killed in the |coprocessor| and the |coprocessor| must guarantee that the related instructions do/did  |
+  |                    |                        | not change architectural state.                                                                                              |
   +--------------------+------------------------+------------------------------------------------------------------------------------------------------------------------------+
 
-The ``commit_valid`` signal will be 1 exactly one ``clk`` cycle for every offloaded instruction by the |coprocessor| (whether accepted or not). The ``id`` value indicates which offloaded
-instruction is allowed to be committed or is supposed to be killed.
+The ``commit_valid`` signal will be 1 exactly one ``clk`` cycle.
+It is not required that a commit transaction is performed for each offloaded instruction individually.
+Instructions can be signalled to be non-speculative or to be killed in batch.
+E.g. signalling the oldest instruction to be killed is equivalent to requesting a flush of the |coprocessor|.
+The first instruction to be considered not-to-be-killed after a commit transaction with ``commit_kill`` as 1,
+is at earliest an instruction with successful issue transaction starting at least one clock cycle later.
+
+.. note::
+
+  If an instruction is marked in the |coprocessor| as killed or committed, the |coprocessor| shall ignore any subsequent commit transaction related to that instruction.
+
+.. note::
 
-For each offloaded and accepted instruction the core is guaranteed to (eventually) signal that such an instruction is either no longer speculative and can be committed (``commit_valid`` is 1
-and ``commit_kill`` is 0) or that the instruction must be killed (``commit_valid`` is 1 and ``commit_kill`` is 1). 
+  A |coprocessor| may receive values of ``commit.id`` that it did not accept.
```

Comment:
or for IDs for which it did not receive a corresponding issue transaction.

Reply:
If a coprocessor did not receive an issue transaction, it definitely did not accept the instruction. So that should be covered.